cpp/tests/transform/row_bit_count_test.cu

```diff
@@ -0,0 +1,576 @@
+  std::unique_ptr<column> expected_sizes;
+  std::tie(struct_col, expected_sizes) = build_struct_column();
+  table_view t({*struct_col});
+  auto result = cudf::row_bit_count(t);
+
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(*expected_sizes, *result);
+}
+
+TEST_F(RowBitCount, StructsNested)
+{
+  // struct<struct<int>, int16>
+  cudf::test::fixed_width_column_wrapper<int> col0{0, 1, 2, 3, 4, 5};
+  cudf::test::structs_column_wrapper inner_struct({col0});
+
+  cudf::test::fixed_width_column_wrapper<int16_t> col1{8, 9, 10, 11, 12, 13};
+  cudf::test::structs_column_wrapper struct_col({inner_struct, col1});
+
+  table_view t({struct_col});
+  auto result = cudf::row_bit_count(t);
+
+  // expect num_rows * (4 + 2) bytes
+  auto size_iter =
+    cudf::detail::make_counting_transform_iterator(0, [&](int i) { return (4 + 2) * CHAR_BIT; });
+  cudf::test::fixed_width_column_wrapper<size_type> expected(size_iter, size_iter + t.num_rows());
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, *result);
+}
+
+std::pair<std::unique_ptr<column>, std::unique_ptr<column>> build_nested_column(
+  std::vector<bool> const& struct_validity)
+{
+  // tests the "branching" case ->  list<struct<list> ...>>>
+
+  // List<Struct<List<int>, float, int16>
+
+  // Inner list column
+  // clang-format off
+  cudf::test::lists_column_wrapper<int> list{
+    {1, 2, 3, 4, 5},     
+    {6, 7, 8},
+    {33, 34, 35, 36, 37, 38, 39},
+    {-1, -2},
+    {-10, -11, -1, -20},
+    {40, 41, 42},
+    {100, 200, 300},
+    {-100, -200, -300}};
+  // clang-format on
+
+  // floats
+  std::vector<float> ages{5, 10, 15, 20, 4, 75, 16, -16};
+  std::vector<bool> ages_validity = {1, 1, 1, 1, 0, 1, 0, 1};
+  auto ages_column =
+    cudf::test::fixed_width_column_wrapper<float>(ages.begin(), ages.end(), ages_validity.begin());
+
+  // int16 values
+  std::vector<int16_t> vals{-1, -2, -3, 1, 2, 3, 8, 9};
+  auto i16_column = cudf::test::fixed_width_column_wrapper<int16_t>(vals.begin(), vals.end());
+
+  // Assemble struct column
+  auto struct_column =
+    cudf::test::structs_column_wrapper({list, ages_column, i16_column}, struct_validity);
+
+  // wrap in a list
+  std::vector<int> outer_offsets{0, 1, 1, 3, 6, 7, 8};
+  cudf::test::fixed_width_column_wrapper<int> outer_offsets_col(outer_offsets.begin(),
+                                                                outer_offsets.end());
+  auto const size = static_cast<column_view>(outer_offsets_col).size() - 1;
+
+  cudf::test::fixed_width_column_wrapper<size_type> expected_sizes{276, 32, 520, 572, 212, 212};
+
+  return {cudf::make_lists_column(static_cast<cudf::size_type>(size),
+                                  outer_offsets_col.release(),
+                                  struct_column.release(),
+                                  cudf::UNKNOWN_NULL_COUNT,
+                                  rmm::device_buffer{}),
+          expected_sizes.release()};
+}
+
+std::unique_ptr<column> build_nested_column2(std::vector<bool> const& struct_validity)
```

Comment:
Could we come up with a better name here?
Perhaps call this one `build_nested_column` and the other one `build_nested_and_expected_column`